test/Microsoft.AspNetCore.Server.KestrelTests/ConnectionTests.cs

```diff
@@ -29,13 +32,20 @@ public void DoesNotEndConnectionOnZeroRead()
                     ServerAddress = ServerAddress.FromUrl("http://127.0.0.1:0"),
                     Thread = engine.Threads[0]
                 };
-                var socket = new MockSocket(mockLibuv, Thread.CurrentThread.ManagedThreadId, trace);
-                var connection = new Connection(context, socket);
-                connection.Start();
 
-                Libuv.uv_buf_t ignored;
-                mockLibuv.AllocCallback(socket.InternalGetHandle(), 2048, out ignored);
-                mockLibuv.ReadCallback(socket.InternalGetHandle(), 0, ref ignored);
+                MockSocket socket = null;
+                Connection connection = null;
+                await context.Thread.PostAsync(_ =>
```

Comment:
This certainly seems more correct, by why did this just start failing with this change?